src/imagetilesource.js

```diff
@@ -0,0 +1,274 @@
+                    source: url
+                });
+            });
+
+            image.src = url;
+        },
+        /**
+         * @function
+         * @param {Number} level
+         */
+        getLevelScale: function (level) {
+            var levelScale = NaN;
+            if (level >= this.minLevel && level <= this.maxLevel) {
+                levelScale =
+                        this.levels[level].width /
+                        this.levels[this.maxLevel].width;
+            }
+            return levelScale;
+        },
+        /**
+         * @function
+         * @param {Number} level
+         */
+        getNumTiles: function (level) {
+            var scale = this.getLevelScale(level);
+            if (scale) {
+                return new $.Point(1, 1);
+            } else {
+                return new $.Point(0, 0);
+            }
+        },
+        /**
+         * @function
+         * @param {Number} level
+         * @param {OpenSeadragon.Point} point
+         */
+        getTileAtPoint: function (level, point) {
+            return new $.Point(0, 0);
+        },
+        /**
+         * This method is not implemented by this class other than to throw an Error
+         * announcing you have to implement it.  Because of the variety of tile
+         * server technologies, and various specifications for building image
+         * pyramids, this method is here to allow easy integration.
+         * @function
+         * @param {Number} level
+         * @param {Number} x
+         * @param {Number} y
+         * @throws {Error}
+         */
+        getTileUrl: function (level, x, y) {
+            var url = null;
+            if (level >= this.minLevel && level <= this.maxLevel) {
+                url = this.levels[level].url;
+            }
+            return url;
+        }
+    });
+
+    function buildLevels(image, minWidth, minHeight) {
+        var levels = [{
+                url: image.src,
+                width: image.naturalWidth,
+                height: image.naturalHeight
+            }];
+
+        var currentWidth = Math.floor(image.naturalWidth / 2);
+        var currentHeight = Math.floor(image.naturalHeight / 2);
+
+        if (currentWidth < minWidth || currentHeight < minHeight) {
+            return levels;
+        }
+
+        var bigCanvas = document.createElement("canvas");
+        var bigContext = bigCanvas.getContext("2d");
+
+        bigCanvas.width = currentWidth;
+        bigCanvas.height = currentHeight;
+        bigContext.drawImage(image, 0, 0, currentWidth, currentHeight);
+
+        if (isCanvasTainted(bigContext)) {
+            // If the canvas is tainted, we can't compute the pyramid.
+            return levels;
+        }
+        levels.splice(0, 0, {
+            url: bigCanvas.toDataURL(),
+            width: currentWidth,
+            height: currentHeight
+        });
+
+        var smallCanvas = document.createElement("canvas");
```

Comment:
This won't work in IE8 either of course. We could still support IE8 with this tile source otherwise, just not the buildPyramid feature.